static/src/js/util/orders.js

```diff
@@ -0,0 +1,55 @@
+import cleanDeep from 'clean-deep'
+import snakeCaseKeys from 'snakecase-keys'
+
+import { prepareGranuleParams, buildGranuleSearchParams } from './granules'
+
+
+/**
+ * Prepare parameters used in submitOrders() based on current Redux State
+ * @param {object} state Current Redux State
+ * @returns Parameters used in submitOrders()
+ */
+export const prepareOrderParams = (state) => {
+  const {
+    authToken,
+    metadata = {},
+    project
+  } = state
+
+  const { collections } = metadata
+  const { byId } = collections
+  const {
+    byId: configById,
+    collectionIds: projectIds
+  } = project
+
+  const projectCollections = []
+  projectIds.forEach((collectionId) => {
+    const projectCollection = byId[collectionId]
+    const { granules } = projectCollection
+
+    const returnValue = {}
+
+    returnValue.id = collectionId
+    returnValue.granule_count = granules.hits
+
+    const params = buildGranuleSearchParams(prepareGranuleParams(state, collectionId))
```

Comment:
This seems to build params with correct casing, change them to javascript casing, then change them back to correct casing for cmr.

Reply:
`prepareGranuleParams` doesn't have them in CMR format, `buildGranuleSearchParams` at least gives them the correct names, then the next two lines clean up the object and snake cases the keys